katdal/visdatav4.py

```diff
@@ -304,12 +303,12 @@ def _before(date):
         # Populate antenna -> receiver mapping and figure out noise diode
         for ant in cam_ants:
             # Try sanitised version of RX serial number first
-            rx_serial = attrs.get('%s_rsc_rx%s_serial_number' % (ant, band), 0)
-            self.receivers[ant] = '%s.%d' % (band, rx_serial)
-            nd_sensor = '%s_dig_%s_band_noise_diode' % (ant, band)
+            rx_serial = attrs.get(f'{ant}_rsc_rx{band}_serial_number', 0)
+            self.receivers[ant] = f'{band}.{rx_serial:d}'
```

Comment:
Is the `:d` actually needed?

Reply:
I think it was a reminder to myself that this is an integer serial number and not some string. So this is a type check of some sort.

Reply:
I don't like crying code, so I'll rather remove it. The line above it shows that `rx_serial` has a default value of 0, which suggests the type already.